src/components/Form.js

```diff
@@ -1,22 +1,41 @@
-import React from 'react'
+import React, { useState } from 'react'
 import { Button } from './Button'
 import { TextInput } from './TextInput'
 import styles from './Form.module.css'
+import { useFormik } from 'formik'
+import { object, string } from 'yup'
+import { Link } from 'react-router-dom'
 
 export const Form = ({ type = 'signup' }) => {
```

Comment:
I think I forgot to mention this the last time. I would give this form a better name and despite the similarities between these forms I would split it into (SignInForm, SignUpForm). They are conceptually different forms and might also change due to different reasons (eg. SignUp requires a Terms And Conditions read flag)